servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/PublisherFlatMapMerge.java

```diff
@@ -198,11 +196,18 @@ public void onSubscribe(final Subscription s) {
         public void onNext(@Nullable final T t) {
             final Publisher<? extends R> publisher = requireNonNull(source.mapper.apply(t),
                     () -> "Mapper " + source.mapper + " returned null");
-            FlatMapPublisherSubscriber<T, R> subscriber = new FlatMapPublisherSubscriber<>(this);
-            if (cancellableSet.add(subscriber) && activeMappedSourcesUpdater.incrementAndGet(this) > 0) {
-                publisher.subscribeInternal(subscriber);
+            for (;;) {
+                final int currValue = this.activeMappedSources;
+                if (currValue < 0) {
+                    throw new IllegalStateException("onNext(" + t + ") after terminal signal delivered to " + this);
```

Comment:
1. `t` may contain sensitive user information, we should not use it as-is for string messages.
2. Consider adding a comment to clarify why we throw from `onNext` instead of delivering `onError`.